can/interfaces/dg/dg.py

```diff
@@ -0,0 +1,456 @@
+
+# ----------------------------------------------------------------------
+import weakref
+from can import BusABC, Message
+from can.interfaces.dg.dg_gryphon_protocol import server_commands
+from can.broadcastmanager import (
+    LimitedDurationCyclicSendTaskABC,
+    ModifiableCyclicTaskABC,
+)
+
+
+class DGBus(BusABC):
+    """ Beacon python-can backend API """
+
+    def __init__(self, channel=1, **kwargs):
+        """
+        :param int channel:
+            Channel which the bus will be initialized on.
+
+        :param bool is_fd:
+            Specifies if bus is CANFD or not.
+
+        :param str ip:
+            IP at which the DG device is connected.
+
+        :param int bitrate:
+            Bitrate for the bus.
+
+        :param bool termination:
+            Termination for the bus (on or off).
+
+        :param int data_bitrate:
+            The data phase bitrate for the bus (Optional, only if is_fd is
+            True).
+
+        :param bool pre_iso:
+            Specifies if bus should be CANFD PREISO (Optional, only if is_fd is
+            True).
+
+        :return: None
+        """
+        self.ip = kwargs["ip"] if "ip" in kwargs else "localhost"
+        self.is_fd = kwargs["is_fd"] if "is_fd" in kwargs else False
+        self.bitrate = kwargs["bitrate"] if "bitrate" in kwargs else 500000
+        self.termination = kwargs["termination"] if "termination" in kwargs else True
+        self.pre_iso = kwargs["pre_iso"] if "pre_iso" in kwargs else False
+        self.data_bitrate = (
+            kwargs["data_bitrate"] if "data_bitrate" in kwargs else 2000000
+        )
+        self.channel = channel
+
+        self.beacon = server_commands.Gryphon(self.ip)
+        self.beacon.CMD_SERVER_REG("root", "dgbeacon")
+        self.beacon.CMD_BCAST_ON()
+        if not self.is_fd:
+            self.mode = "CAN"
+            temp_mode = [0]
+        elif self.pre_iso:
+            self.mode = "CANPREISO"
+            temp_mode = [2]
+        else:
+            self.mode = "CANFD"
+            temp_mode = [1]
+        temp_databitr = DGBus._int_to_list(self.data_bitrate)
+        temp_databitr.reverse()
+        self.beacon.CMD_CARD_IOCTL(
+            self.channel, self.beacon.IOCTL_GSETFASTBITRATE, data_in=temp_databitr
+        )
+
+        temp_bitr = DGBus._int_to_list(self.bitrate)
+        temp_bitr.reverse()
+        temp_term = [1] if self.termination else [0]
+
+        self.beacon.CMD_CARD_IOCTL(
+            self.channel, self.beacon.IOCTL_SETINTTERM, data_in=temp_term
+        )
+        self.beacon.CMD_CARD_IOCTL(
+            self.channel, self.beacon.IOCTL_GSETBITRATE, data_in=temp_bitr
+        )
+        self.beacon.CMD_CARD_IOCTL(
+            self.channel, self.beacon.IOCTL_GCANSETMODE, data_in=temp_mode
+        )
+        self.beacon.CMD_INIT(self.channel, value_in=self.beacon.ALWAYS_INIT)
+        self.beacon.CMD_CARD_SET_FILTER_MODE(
+            self.channel, self.beacon.FILTER_OFF_PASS_ALL
+        )
```

Comment:
I assume there isn't any downside to calling this, and then a few lines down via the `super` calling `set_filters`?